src/aiida/orm/querybuilder.py

```diff
@@ -690,16 +715,23 @@ def add_filter(self, tagspec: Union[str, EntityClsType], filter_spec: FilterType
     @staticmethod
     def _process_filters(filters: FilterType) -> Dict[str, Any]:
```

Comment:
The repurposing of `filters` is not appreciated by `mypy`. I went around this using `filters_dict` but still had to add `type: ignore`s to deal with overwrites. I suspect there is a "correct" or at least "better" approach to handling this situation.